s5_base/run.sh

```diff
@@ -104,29 +101,53 @@ fi
 
 # train a first delta + delta-delta triphone(tri1) system on all utterances
 if [ $stage -le 6 ]; then
+  echo "Train deltas triphone1 model"
   steps/train_deltas.sh --boost-silence 1.25 --cmd "$train_cmd" \
     2000 10000 data/train data/lang exp/mono_ali_train exp/samromur_tri1
-fi
 
-exit 0
-# TODO: before decoding create a tgmed or tglarge arpalm
-if [ $stage -eq 7 ]; then
   # decode using the tri1 model
+  echo "Create deltas decoding graph"
   (
     utils/mkgraph.sh data/lang_3gsmall \
       exp/samromur_tri1 exp/samromur_tri1/graph_3gsmall
     for test in dev; do
-      steps/decode.sh --nj 5 --cmd "$decode_cmd" exp/samromur_tri1/graph_tgsmall \
+      steps/decode.sh --nj 5 --cmd "$decode_cmd" exp/samromur_tri1/graph_3gsmall \
       data/$test exp/samromur_tri1/decode_3gsmall_$test
-      steps/lmrescore.sh --cmd "$decode_cmd" data/lang_{3gsmall,tgmed} \
-        data/$test exp/samromur_tri1/decode_{3gsmall,tgmed}_$test
+      steps/lmrescore.sh --cmd "$decode_cmd" data/lang_{3gsmall,3gmed} \
+        data/$test exp/samromur_tri1/decode_{3gsmall,3gmed}_$test
       steps/lmrescore_const_arpa.sh \
-        --cmd "$decode_cmd" data/lang_{3gsmall,tglarge} \
-        data/$test exp/samromur_tri1/decode_{3gsmall,tglarge}_$test
+        --cmd "$decode_cmd" data/lang_{3gsmall,3glarge} \
+        data/$test exp/samromur_tri1/decode_{3gsmall,3glarge}_$test
     done
   )&
 
   steps/align_si.sh --nj 5 --cmd "$train_cmd" \
     data/train data/lang exp/samromur_tri1 exp/tri1_ali_train
 fi
 
+# train an LDA+MLLT system triphone2.
+if [ $stage -le 7 ]; then
+  steps/train_lda_mllt.sh --cmd "$train_cmd" \
+    --splice-opts "--left-context=3 --right-context=3" 2500 15000 \
+    data/train data/lang exp/tri1_ali_train exp/samromur_tri2b
+
+  # decode using the LDA+MLLT model
+  (
+    utils/mkgraph.sh data/lang_3gsmall \
+      exp/samromur_tri2b exp/samromur_tri2b/graph_3gsmall
+    for test in dev; do
+      steps/decode.sh --nj 10 --cmd "$decode_cmd" exp/samromur_tri2b/graph_3gsmall \
+        data/$test exp/samromur_tri2b/decode_3gsmall_$test
+      steps/lmrescore.sh --cmd "$decode_cmd" data/lang_{3gsmall,3gmed} \
+        data/$test exp/samromur_tri2b/decode_{3gsmall,3gmed}_$test
+      steps/lmrescore_const_arpa.sh \
+        --cmd "$decode_cmd" data/lang_{3gsmall,3glarge} \
+        data/$test exp/samromur_tri2b/decode_{3gsmall,3glarge}_$test
+    done
+  )&
+
+  # Align utts using the tri2b model
+  steps/align_si.sh  --nj 5 --cmd "$train_cmd" --use-graphs true \
+    data/train data/lang exp/samromur_tri2b exp/samromur_tri2b_ali_train
+fi
+
```

Comment:
Just some wonderings. Why do you use a subset of train only for mono training instead of using subsets of increasing size for all of the trainings? Why do you boost silence in train_deltas and train_lda_mllt? And did you try different number of leaves and gaussians in train_lda_mllt or are these just what people generally uses?

Reply:
Well I use it for mono because I wanted to make sure the initial alignments are definitely right. I guess I didn't care as much after that. But I guess I could gradually increase it from 500/mono 20,000/tri1 40,000/tri2 train/tri2b. What do you think of these numbers? The later ones might need to be adjusted depending on the dataset size. 

I don't boost silence in train_lda_mllt (not that I can see at least). I boost it in mono so that the silence mapping is greedier and hopefuly the resulting alignments are more accurate. David did say there are a lot of silences at the beginning and end of samromur. I guess I don't need to do it in train_deltas, too though. Should I take it out?

I haven't tried different numbers of leaves or gaussians so if we have a terrible wer, that would definitely be a place to start experimenting.

Reply:
Mostly I'm just curious about this. I think it is fine to keep the silence boosting in. Using subdirs for these first steps is easy to do and quick so I think that would be good and fine the way you propose, except I find 500 rather small. I think some Kaldi recipes use 5000 for mono, but something there in between if probably good